patroni/postgresql/slots.py

```diff
@@ -240,31 +357,57 @@ def _drop_incorrect_slots(self, cluster: Cluster, slots: Dict[str, Any], paused:
                     self._schedule_load_slots = True
 
     def _ensure_physical_slots(self, slots: Dict[str, Any]) -> None:
+        """Create any missing physical replication *slots*.
+
+        Any failures are logged and do not interrupt creation of all *slots*.
+
+        :param slots: A dictionary mapping slot name to slot attributes. This method only considers a slot
+                      if the value is a dictionary with the key ``type`` and a value of ``physical``.
+        """
         immediately_reserve = ', true' if self._postgresql.major_version >= 90600 else ''
         for name, value in slots.items():
             if name not in self._replication_slots and value['type'] == 'physical':
                 try:
-                    self._query(("SELECT pg_catalog.pg_create_physical_replication_slot(%s{0})"
-                                 " WHERE NOT EXISTS (SELECT 1 FROM pg_catalog.pg_replication_slots"
-                                 " WHERE slot_type = 'physical' AND slot_name = %s)").format(
-                                     immediately_reserve), name, name)
+                    self._query(f"SELECT pg_catalog.pg_create_physical_replication_slot(%s{immediately_reserve})"
+                                f" WHERE NOT EXISTS (SELECT 1 FROM pg_catalog.pg_replication_slots"
+                                f" WHERE slot_type = 'physical' AND slot_name = %s)",
+                                name, name)
                 except Exception:
                     logger.exception("Failed to create physical replication slot '%s'", name)
                 self._schedule_load_slots = True
 
     @contextmanager
-    def get_local_connection_cursor(self, **kwargs: Any) -> Generator[Union['cursor', 'Cursor[Any]'], None, None]:
+    def get_local_connection_cursor(self, **kwargs: Any) -> Iterator[Union['cursor', 'Cursor[Any]']]:
```

Comment:
why did you change the return type here but not here?
https://github.com/zalando/patroni/blob/be8ad58afbfab90b3cc190d541ae3564c6c49554/patroni/postgresql/slots.py#L525

and maybe also here then
https://github.com/zalando/patroni/blob/b7caf3b7f23924711766ef24250d09c4c198b4a1/patroni/postgresql/connection.py#L47

Reply:
> why did you change the return type here but not here?

That was Israel's suggestion, so I missed add it there. I'll fix that.

> and maybe also here then
> 
> https://github.com/zalando/patroni/blob/b7caf3b7f23924711766ef24250d09c4c198b4a1/patroni/postgresql/connection.py#L47

Perhaps in another PR?

Reply:
> Perhaps in another PR?

Yes, sure. Just hope we will not forget:)

Reply:
Yeah I had already ;-)

https://github.com/zalando/patroni/pull/2799